prow/logrusutil/logrusutil.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package logrusutil implements some helpers for using logrus
+package logrusutil
+
+import (
+	"github.com/sirupsen/logrus"
+)
+
+// DefaultFieldsFormatter wraps another logrus.Formatter, injecting
+// DefaultFields into each Format() call, existing fields are preserved
+// if they have the same key
+type DefaultFieldsFormatter struct {
+	WrappedFormatter logrus.Formatter
```

Comment:
We could make this default to the JSON formatter if it is nil.

Reply:
OK how about we have `NewDefaultFieldsFormatter(wrapped *logrus.Formatter, default logrus.Fields)` and default to json if wrapped is nil?

Reply:
done, switched to this everywhere